lib/client.js

```diff
@@ -0,0 +1,635 @@
+
+    this.dcIndex = -1;
+    this.multiDC = opts.multiDC === undefined ? true : opts.multiDC;
+    this.datacenters = null;
+
+    if (opts.pollInterval !== undefined && !opts.standalone) {
+        this.pollInterval = (opts.pollInterval * 1000);
+        this.defMaxSizeMB = opts.defaultMaxStreamingSizeMB ||
+            DEF_MAX_STREAMING_SIZE_MB;
+        this.utilization = opts.maxUtilizationPct ||
+            DEF_MAX_PERCENT_UTIL;
+        setImmediate(doPoll.bind(this));
+    }
+}
+util.inherits(StorinfoClient, EventEmitter);
+
+StorinfoClient.prototype.close = function close() {
+    if (this.pollTimer) {
+        clearTimeout(this._storageTimer);
+    }
+    if (!this.standalone) {
+        this.http.close();
+    }
+};
+
+/*
+ * Get the data for a specific row (storage node) from the manta_storage bucket.
+ * This takes a single required argument, which is the manta_storage_id of the
+ * storage node.
+ *
+ * On success, the "err" parameter to the callback will be null and the "obj"
+ * param will contain manta_storage object for the requested storage node.
+ *
+ * On failure, the "obj" param will be set to null.
+ */
+StorinfoClient.prototype.pollSpecific = function pollSpecific(storageid,
+    callback) {
+
+    assert.string(storageid, 'storageid');
+
+    /*
+     * pollSpecific isn't supported in standalone mode
+     */
+    if (this.standalone) {
+        var standErr = new errors.StandaloneModeError();
+        callback(null, standErr);
+    }
+
+    this.http.get('/poll?only_id=' + storageid, function (err, req, res, obj) {
+        if (err) {
+            callback(null, err);
+        }
+        callback(obj, null);
+    });
+};
+
+/*
+ * This function wraps the /poll API provided by the Manta Storinfo service.
+ * It takes two arguemnts:
+ *   args: object containing the following properties
+ *       uriParams: optional object containing the following optional params:
+ *           after_id: String manta_storage_id
+ *           limit: Number
+ *
+ *       The args object is also used internally to store the intermiediate and
+ *       final results of successive calls to /poll.
+ *
+ *   callback: callback function that will take two parameters
+ *       res: on success, this will contain the response from /poll
+ *            on error, this will be null
+ *       err: object containing error details if an error occurs
+ *            on success, this param will be null
+ */
+StorinfoClient.prototype.poll = function poll(args, callback) {
+    assert.object(args, 'args');
+    assert.optionalObject(args.uriParams);
+
+    var self = this;
+
+    /*
+     * poll isn't supported in standalone mode
+     */
+    if (self.standalone) {
+        var standErr = new errors.StandaloneModeError();
+        callback(null, standErr);
+    }
+
+    var query = uri.buildQuery(args.uriParams);
+    if (args.res === undefined) {
+        args.res = [];
+    }
+
+    this.http.get('/poll?' + query, function (err, req, res, obj) {
```

Comment:
The restify-clients docs(https://github.com/restify/clients#api-options)  don't make it clear at all, but you can just pass in query params like this:

```
this.http.get({path: '/poll', query: {foo: 'bar', baz: 'bling'}, function onGet(err, req, res, obj) {
    ...
});
```

So I'd suggest:
- drop the "urijs" dep
- do one of the following:

```
    var reqOpts = {
        path: '/poll',
        query: args.uriParams
    };
    this.http.get(reqOpts, function onGet(err, req, res, obj) {
        //...
    };
```

or with more client-side arg handling:

```
    assert.object(opts, 'opts');
    assert.optionalString(opts.after_id, 'opts.after_id');
    assert.optionalString(opts.only_id, 'opts.only_id');
    assert.optionalFinite(opts.limit, 'opts.limit');
    
    var reqOpts = {
        path: '/poll'
        query: {}
    };
    if (opts.after_id) {
        req.query.after_id = opts.after_id;
    }
    if (opts.only_id) {
        req.query.only_id = opts.only_id;
    }
    if (opts.limit) {
        req.query.limit = opts.limit;
    }
    this.http.get(reqOpts, function onGet(err, req, res, obj) {
        //...
    };
```


#### Larger potential naming changes

This is all just opinion and optional -- I realize I'm suggesting just adding work.

It generally would be nice if the client method names matched the names of the API methods as documented. The current Storinfo docs (https://github.com/joyent/manta-storinfo/blob/master/docs/index.md) call this method "GetSharks". If possible usage of "shark" should disappear from code: instead trying to limit to just "storage node" or "storage id" (rather than a mix of vars using alternately "shark" and "storage node" and "mako" that exists in code now).  Say Storinfo API changed the name to "ListStorageNodes", then the client should consider calling the method `client.listStorageNodes`. If you go that route, I'd consider changing from `/poll` to `/storagenodes` as the URL path.

Reply:
I've  changed /poll to /storagenodes and the method name to "GetStorageNodes".

In node-storinfo, I changed StorinfoClient.prototype.poll to StorinfoClient.prototype.getStorageNodes